lib/diff.ml

```diff
@@ -250,8 +299,13 @@ and modified_variant_type ~typing_env ~ref_constructor_lst ~cur_constructor_lst
   in
   modified_cstrs
 
-and diff_list (diff_one : 'a option -> 'a option -> ('a, 'diff) maybe_changed)
-    (ref : 'a list) (curr : 'a list) : ('a, 'diff) maybe_changed list =
+and diff_list :
+      'a 'diff.
+      ('a option -> 'a option -> ('a, 'diff) maybe_changed) ->
```

Comment:
Any particular reason why you removed the arguments labels here?

I think it would be best to keep the labels here!